src/lib/component/audio_post.dart

```diff
@@ -0,0 +1,247 @@
+import 'package:amplify_flutter/amplify_flutter.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+class AudioPost extends StatelessWidget {
+  const AudioPost({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final items = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+    return ListView.separated(
+      itemCount: items.length + 1,
+      separatorBuilder: (context, index) {
+        return const Divider();
+      },
+      itemBuilder: (context, index) {
+        final itemState = Provider.of<ItemState>(context);
+
+        if (index < items.length) {
+          return Container(
+            padding: const EdgeInsets.all(8.0),
+            child: Row(
+              children: [
+                const Avatar(),
+                const Content(),
+                IconButton(
+                  icon: itemState.isPlaying(index) ? const Icon(Icons.pause_circle_filled) : const Icon(Icons.play_circle_filled),
+                  color: const Color.fromRGBO(124, 122, 122, 1.0),
+                  iconSize: 40,
+                  onPressed: () => itemState.updatePressed(index)
+                )
+              ],
+            )
+          );
+        } else {
+            return const SizedBox(height: 100.0); // 最後の要素に到達した際にMyMusisPlayerの高さ分要素を追加
+        }
+      },
+    );
+  }
+}
+
+class ItemState extends ChangeNotifier {
+  final Map<int, bool> _playedMap = {};
+
+  bool isPlaying(int index) {
+    return _playedMap[index] ?? false;
+  }
+
+  void updatePressed(int index) {
+    _playedMap[index] = !isPlaying(index);
+    _playedMap.forEach((key, value) {
+      if(key != index) {
+        _playedMap[key] = false;
+      }
+    });
```

Comment:
[imo]
一度全て`false`にしてしまうようにするとif文を設ける必要なくなりそうです。
```suggestion
    _playedMap.forEach((key, value) {
      _playedMap[key] = false;
    });
    _playedMap[index] = !isPlaying(index);
```

Reply:
ありがとうございます。
修正いたしました。https://github.com/gizumo-oss/yamabico/pull/39/commits/4777e7a2e023199c5a0be3ba3f3f04da1dcd7d1b

Reply:
確かに！と思って[修正しちゃった](https://github.com/gizumo-oss/yamabico/pull/39/commits/4777e7a2e023199c5a0be3ba3f3f04da1dcd7d1b)んですけど、
なんでこうしたのか考えたら、同じボタンの場合の考慮だったのでリバートしました。https://github.com/gizumo-oss/yamabico/pull/39/commits/fe21e2980a5404abbfe23abe99500672bb611e88